egs/aishell/s10/local/run_chain.sh

```diff
@@ -159,57 +164,95 @@ log_level=info # valid values: debug, info, warning
-    --frames-per-eg $frames_per_eg \
-    --frames-per-iter $frames_per_iter \
-    --generate-egs-scp true \
     --left-context $egs_left_context \
-    --left-context-initial -1 \
-    --left-tolerance 5 \
     --right-context $egs_right_context \
-    --right-context-final -1 \
-    --right-tolerance 5 \
-    --srand 0 \
-    --stage -10 \
-    $train_data_dir \
-    $dir $lat_dir $dir/egs 
+    --frame-subsampling-factor $frame_subsampling_factor \
+    --alignment-subsampling-factor $frame_subsampling_factor \
+    --frames-per-chunk $frames_per_eg \
+    --feat-type $feat_type \
+    ${train_data_dir} ${dir} ${lat_dir} ${dir}/raw_egs
 fi
 
+if [ $stage -le 15 ]; then
+  echo "$0: about to process egs"
+  steps/chain2/process_egs.sh  --cmd "$train_cmd" \
+      --num-repeats 1 \
+    ${dir}/raw_egs ${dir}/processed_egs
+fi
+
+if [ $stage -le 16 ]; then
+  echo "$0: about to randomize egs"
+  steps/chain2/randomize_egs.sh --frames-per-job 3000000 \
+    ${dir}/processed_egs ${dir}/egs
+fi
+
+lda_mat_filename=
+if [[ "$feat_type" == "lda" ]]; then
+  if [[ $stage -le 17 ]]; then
+    echo $dir
+    mkdir -p $dir/configs
+    if $train_ivector; then
+      cat <<EOF > $dir/configs/init.config
+input-node name=ivector dim=100
+input-node name=input dim=120
+output-node name=output input=Append(input, ReplaceIndex(ivector, t, 0))
+EOF
+    else
+      cat <<EOF > $dir/configs/init.config
+input-node name=input dim=120
+output-node name=output input=Append(input)
+EOF
+    fi
+    $train_cmd $dir/log/nnet_init.log \
+      nnet3-init --srand=-2 $dir/configs/init.config $dir/configs/init.raw
+
+    echo "$0: Training pre-conditioning matrix"
+    num_lda_jobs=`find ${dir}/egs/ -iname 'train.*.scp' | wc -l | cut -d ' ' -f2`
+    steps/chain2/compute_preconditioning_matrix.sh --cmd "$train_cmd" \
+        --nj $num_lda_jobs \
+        $dir/configs/init.raw \
+        $dir/egs \
+        $dir || exit 1
+  fi
+  lda_mat_filename=$dir/lda.mat
+fi
 
-if [[ $stage -le 16 ]]; then
+merged_egs_dir=merged_egs_chain2
+if [[ $stage -le 19 ]]; then
   echo "$0: merging egs"
-  mkdir -p $dir/merged_egs
-  num_egs=$(ls -1 $dir/egs/cegs*.ark | wc -l)
 
-  $train_cmd --max-jobs-run $nj JOB=1:$num_egs $dir/merged_egs/log/merge_egs.JOB.log \
-    nnet3-chain-shuffle-egs ark:$dir/egs/cegs.JOB.ark ark:- \| \
+  mkdir -p $dir/$merged_egs_dir
+  num_egs=$(ls -1 $dir/egs/train.*.scp | wc -l)
+
+  $train_cmd --max-jobs-run $nj JOB=1:$num_egs $dir/$merged_egs_dir/log/merge_egs.JOB.log \
+    nnet3-chain-shuffle-egs scp:$dir/egs/train.JOB.scp ark:- \| \
     nnet3-chain-merge-egs --minibatch-size=$minibatch_size ark:- \
-      ark,scp:$dir/merged_egs/cegs.JOB.ark,$dir/merged_egs/cegs.JOB.scp || exit 1
+      ark,scp:$dir/$merged_egs_dir/cegs.JOB.ark,$dir/$merged_egs_dir/cegs.JOB.scp || exit 1
 
-  rm $dir/egs/cegs.*.ark
+  rm $dir/raw_egs/cegs.*.ark
 fi
 
-feat_dim=$(cat $dir/egs/info/feat_dim)
+
+feat_dim=$(cat $dir/raw_egs/info.txt | grep 'feat_dim' | awk '{print $NF}')
```

Comment:
Define `info_file=$dir/raw_egs/info.txt` and use it below may be better.

Reply:
Done